packages/shared/src/components/comments/MainComment.tsx

```diff
@@ -66,15 +69,25 @@ export default function MainComment({
   const { commentId, inputProps, onReplyTo } = useComments(props.post);
   const { inputProps: editProps, onEdit } = useCommentEdit();
 
+  const initialInView = !lazy;
+  const { ref: inViewRef, inView } = useInView({
+    triggerOnce: true,
+    initialInView,
+  });
+
   return (
     <section
+      ref={inViewRef}
       className={classNames(
         'flex scroll-mt-16 flex-col items-stretch rounded-24 border border-theme-divider-tertiary',
         className?.container,
       )}
       data-testid="comment"
+      style={{
+        contentVisibility: initialInView ? 'visible' : 'auto',
+      }}
```

Comment:
This additionally allows better scroll performance since views not visible will not be rendered.

Reply:
sadness tailwind doesn't support it yet.
https://github.com/tailwindlabs/tailwindcss/discussions/2162

But yeah nice thinking, totally didn't think about this.